### ISSUE TITLE
Error during metrics collection results in 500

### ISSUE THREAD

**Issue by OlivierBoucher** [frodenas/stackdriver_exporter#9]
Hi,

As soon as I added `logging.googleapis.com`  to the prefixes it started to crash.

Here is the config:

``` 
[
    {
        "name": "STACKDRIVER_EXPORTER_GOOGLE_PROJECT_ID",
        "value": "eflow-dev"
    },
    {
        "name": "STACKDRIVER_EXPORTER_MONITORING_METRICS_TYPE_PREFIXES",
        "value": "cloudsql.googleapis.com,datastore.googleapis.com,pubsub.googleapis.com,bigquery.googleapis.com,logging.googleapis.com"
    },
    {
        "name": "STACKDRIVER_EXPORTER_WEB_LISTEN_ADDRESS",
        "value": ":8080"
    },
    {
        "name": "STACKDRIVER_EXPORTER_MONITORING_METRICS_INTERVAL",
        "value": "5m"
    }
]
```

The error seems to come from prometheus itself and only gets printed when you query /metrics

```
An error has occurred during metrics collection:

138 error(s) occurred:
* collected metric stackdriver_monitoring_logging_googleapis_com_log_entry_count label:<name:"cluster_name" value:"benchmark-cluster" > label:<name:"container_name" value:"j-clicker" > label:<name:"instance_id" value:"272634823886272651" > label:<name:"log" value:"j-clicker" > label:<name:"namespace_id" value:"default" > label:<name:"pod_id" value:"j-clicker-vvwnw" > label:<name:"project_id" value:"eflow-dev" > label:<name:"resource_type" value:"gke_container" > label:<name:"severity" value:"INFO" > label:<name:"unit" value:"1" > label:<name:"zone" value:"us-east1-c" > counter:<value:0 >  has label dimensions inconsistent with previously collected metrics in the same metric family 
... and so on
```

**Comment by frodenas** [comment]
@OlivierBoucher Thanks for reporting this. I've identified the problem but I'm still trying to find the right solution for this.

When the exporter is reporting metrics it adds several labels, the ones that came from the [metric](https://github.com/frodenas/stackdriver_exporter/blob/master/collectors/monitoring_collector.go#L262) itself, and the ones that came from the [monitored resource](https://github.com/frodenas/stackdriver_exporter/blob/master/collectors/monitoring_collector.go#L266). Usually the monitored resource is the same (because metrics are namespaced), but `logging.googleapis.com` is a bit special, because the monitored resource might be different. This means that depending on the resource emitting logs, the labels will differ, and hence the `has label dimensions inconsistent with previously collected metrics` error.

**Comment by frodenas** [comment]
This has been fixed at [v0.2.0](https://github.com/frodenas/stackdriver_exporter/releases/tag/v0.2.0). Please be warned that metric's names have changed in this release (ie, the above metric has changed from`stackdriver_monitoring_logging_googleapis_com_log_entry_count` to `stackdriver_gke_container_logging_googleapis_com_log_entry_count`, note that instead of `monitoring` now it has the resource type `gke_container`).

**Comment by OlivierBoucher** [comment]
@frodenas Thank you very much
